fabric-registry-sync-v0/src/main/java/net/fabricmc/fabric/impl/registry/sync/map/IdMap.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.registry.sync.map;
+
+import java.util.Comparator;
+import java.util.List;
+
+import it.unimi.dsi.fastutil.objects.Object2IntLinkedOpenHashMap;
+
+import net.minecraft.util.Identifier;
+
+public class IdMap extends Object2IntLinkedOpenHashMap<Identifier> {
+	/**
+	 * Returns an entry pair list sorted by the rawId.
+	 */
+	public List<Entry<Identifier>> sortedEntryList() {
+		return object2IntEntrySet().stream()
+				.sorted(Comparator.comparingInt(Entry::getIntValue))
+				.toList();
+	}
```

Comment:
I don't think it is a good idea to extend collection classes. This could just as easily be a method somewhere else that takes the ID map as a parameter.

Reply:
agreed